qiskit/transpiler/_transpiler.py

```diff
@@ -88,92 +79,83 @@ def compile(circuits, backend,
             not all(key in hpc for key in ('multi_shot_optimization', 'omp_num_threads')):
         raise QISKitError('Unknown HPC parameter format!')
 
+    # step 3: populate the `circuits` in qobj, after compiling each circuit
     qobj['circuits'] = []
     if not basis_gates:
-        if 'basis_gates' in backend_conf:
-            basis_gates = backend_conf['basis_gates']
-    if len(basis_gates.split(',')) < 2:
```

Comment:
What about this case? Is it not necessary to check if the base is only one gate in order to provide a better set?

Reply:
no this was just a hack to catch some `basis_gate` specifications over the API that didn't conform to the standard. now all is fixed in the API, so this can go.

Reply:
no this was just a temporary hack to catch deprecated ways of specifying `basis_gates`, while they were not corrected in the API. now all is correct, and we can retire this.